setuptools_rust/extension.py

```diff
@@ -280,46 +284,45 @@ class RustBin(RustExtension):
             `the Cargo Book <https://doc.rust-lang.org/cargo/commands/cargo-build.html#manifest-options>`_.
             For example, ``cargo_manifest_args=["--locked"]`` will require
             ``Cargo.lock`` files are up to date.
-        features: A list of Cargo features to also build.
-        rustc_flags: A list of additional flags passed to rustc.
-        rust_version: Minimum Rust compiler version required for this
-            extension.
+        features: Cargo `--features` to add to the build.
+        rust_version: Minimum Rust compiler version required for this bin.
         quiet: Suppress Cargo's output.
         debug: Controls whether ``--debug`` or ``--release`` is passed to
             Cargo. If set to `None` (the default) then build type is
             automatic: ``inplace`` build will be a debug build, ``install``
             and ``wheel`` builds will be release.
         strip: Strip symbols from final file. Does nothing for debug build.
-        native: Build extension or executable with ``--target-cpu=native``.
+        optional: If it is true, a build failure in the bin will not
+            abort the build process, and instead simply not install the failing
+            bin.
     """
 
     def __init__(
         self,
-        target: str,
+        target: Union[str, Dict[str, str]],
         path: str = "Cargo.toml",
-        args: Optional[List[str]] = None,
-        cargo_manifest_args: Optional[List[str]] = None,
-        features: Optional[List[str]] = None,
-        rustc_flags: Optional[List[str]] = None,
```

Comment:
Deliberately removed `rustc_flags` from `RustBin` because it uses `cargo build` so never passes these flags.